File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        int[] init = new int[] {0, 1, 1};
        int[] fibonacci = new int[n + 1];
        
        if (n < 3) {
            return init[n];
        } else {
            for (int i = 0; i < 3; i++) {
                fibonacci[i] = init[i];
            }
            
            for (int i = 3; i <= n; i++) {
                fibonacci[i] = fibonacci[i - 1] + fibonacci[i - 2] + fibonacci[i - 3];
            }
            
            return fibonacci[n];
        }
    }
}

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
  map<int,int>dp;
  int dfs(int n){
    if(n==0){
      return 0;
    }else if(n==1){
      return 1;
    }else if(n==2){
      return 1;
    }
    if(dp.find(n)!=dp.end()){
      return dp[n];
    }
    return dp[n]=dfs(n-1)+dfs(n-2)+dfs(n-3);
  }
  int tribonacci(int n) {
    return dfs(n);
  }
};